java/code/src/com/redhat/rhn/manager/session/SessionManager.java

```diff
@@ -86,7 +86,7 @@ public static WebSession makeSession(Long uid, long duration) {
 
         s.setExpires(TimeUtils.currentTimeSeconds() + duration);
         WebSessionFactory.save(s);
-        HibernateFactory.getSession().flush();
+        HibernateFactory.commitTransaction();
```

Comment:
This should be done in the servlet filter handling the transaction. Do you know why we need to add it here? It could be dangerous to close the transaction earlier if some other code uses the transaction after the commit.

Reply:
It would indeed be cleaner to have spark go through the filter too. If not possible then merge